File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
  public static double SuccessRate(int speed)
  {
    if (speed == 0) return 0;
    else if (speed <= 4) return 1.0;
    else if (speed <= 8) return .90;
    else if (speed == 9) return .80;
    else return .77;
  }

  public static double ProductionRatePerHour(int speed)
  {
    return speed * 221 * SuccessRate(speed);
  }

  public static int WorkingItemsPerMinute(int speed)
  {
    return (int)Math.Floor(ProductionRatePerHour(speed) / 60);
  }
}

File: solutions/fsharp/cars-assemble/1/CarsAssemble.fs
module CarsAssemble

let CARS_PER_HOUR_PER_SPEED = 221

let successRate (speed: int) : float =
    match speed with
    | 0 -> 0.0
    | 1
    | 2
    | 3
    | 4 -> 1.0
    | 5
    | 6
    | 7
    | 8 -> 0.9
    | 9 -> 0.8
    | 10 -> 0.77
    | _ -> failwith "invalid speed"

let productionRatePerHour (speed: int) : float =
    float speed * float CARS_PER_HOUR_PER_SPEED * (successRate speed)

let workingItemsPerMinute (speed: int) : int =
    int ((productionRatePerHour speed) / float 60)
